jsm.go

```diff
@@ -161,6 +161,37 @@ type APIError struct {
 	Description string `json:"description,omitempty"`
 }
 
+// JetStreamAPIError is an error result from making a request to the
+// JetStream API.
+type JetStreamAPIError interface {
+	Code() int
+	ErrorCode() int
+	Description() string
+	Error() string
```

Comment:
could add some other conditions to be able signal the client that something is `Temporary()` here too, for example when there are transient errors such as `JSClusterNotAvailable`